qcodes/instrument_drivers/tektronix/Keithley_2600_channels.py

```diff
@@ -190,6 +191,65 @@ def get_raw(self) -> np.ndarray:  # pylint: disable=E0202
         return np.linspace(0, dt*npts, npts, endpoint=False)
 
 
+class _ParameterWithStatus(Parameter):
+    def __init__(self, *args: Any, **kwargs: Any):
+        super().__init__(*args, **kwargs)
+
+        self._measurement_status: Optional[str] = None
+
+    @property
+    def measurement_status(self) -> Optional[str]:
+        return self._measurement_status
+
+    def snapshot_base(self, update: Optional[bool] = True,
+                      params_to_skip_update: Optional[Sequence[str]] = None
+                      ) -> Dict[Any, Any]:
+        snapshot = super().snapshot_base(
+            update=update, params_to_skip_update=params_to_skip_update
+        )
+
+        if self._snapshot_value:
+            snapshot["measurement_status"] = self.measurement_status
+
+        return snapshot
+
+
+class _MeasurementCurrentParameter(_ParameterWithStatus):
+
+    def __init__(self, *args: Any, **kwargs: Any):
+        super().__init__(*args, **kwargs)
+
+    def _meas_status(self) -> None:
+        assert isinstance(self.instrument, KeithleyChannel)
+
+        smu = self.instrument
+        channel = self.instrument.channel
+
+        meas_status = smu.ask(f'status.measurement.instrument.'
+                              f'{channel}.enable = measurementRegister')
+
+        status_bits = [int(i) for i in bin(
+            int(meas_status)
+        ).replace('0b', '').zfill(16)[::-1]]
+
+        if status_bits[1]:
+            self._measurement_status = 'Current compliance is hit.'
+        else:
+            self._measurement_status = 'Measurement status normal.'
+
+    def get_raw(self) -> ParamRawDataType:
+        assert isinstance(self.instrument, KeithleyChannel)
+
+        smu = self.instrument
+        channel = self.instrument.channel
+
+        value = float(smu.ask(f'{channel}.measure.i()'))
+
+        self._meas_status()
```

Comment:
This means that every time we get a value from the instrument we will have to ask it twice. This seems to likely be a significant slowdown. Is there a way to either get this from a status register or perhaps ask about both at the same time e.g. perhaps one can chain the commands ?

Reply:
Ok, yes, I'll check the possibility of chaining the commands.